src/libraries/System.Private.CoreLib/src/System/Reflection/Assembly.cs

```diff
@@ -255,7 +255,7 @@ public static Assembly LoadFile(string path)
                 throw new ArgumentException(SR.Format(SR.Argument_AbsolutePathRequired, path), nameof(path));
             }
 
-            string normalizedPath = Path.GetFullPath(path);
+            string normalizedPath = Path.GetFullNonEmptyPath(path);
```

Comment:
This avoids redundant check for non-null non-empty path that is very cheap. This check costs nothing compared to the cost of the rest of the API and the overall cost of this method. We live with the redundant cheap check checks like this in many (thousands) of place in this repo. It is not worth it to try to avoid them by making the code more complex.

The focus of this PR should be on removing the redundant `Contains('\0')` checks that are somewhat expensive, not on removing the redundant checks for non-null non-empty paths.

Reply:
I reverted changes and refocused this PR on the proper changes. Thank you so much for your feedback.